src/test/java/com/puppycrawl/tools/checkstyle/XMLLoggerTest.java

```diff
@@ -398,6 +400,25 @@ public void testAuditFinishedWithoutFileFinished() throws IOException {
         verifyLines(expectedLines);
     }
 
+    @Test
+    public void testClearsMessagesAfterAuditFinished() throws IOException {
+        final XMLLogger logger = new XMLLogger(outStream, true);
+        logger.auditStarted(null);
+        final AuditEvent fileStartedEvent = new AuditEvent(this, "Test.java");
+        logger.fileStarted(fileStartedEvent);
+
+        final LocalizedMessage message =
+                new LocalizedMessage(1, 1,
+                        "messages.properties", "key", null, SeverityLevel.ERROR, null,
+                        getClass(), null);
+        final AuditEvent errorEvent = new AuditEvent(this, "Test.java", message);
+        logger.addError(errorEvent);
+        logger.auditFinished(null);
+
+        final Map<String, ?> messagesMap = Whitebox.getInternalState(logger, "fileMessages");
```

Comment:
instead of using reflection , please use the same instance of logger to "process" another file and make sure amount of violations in output is 1 for both runs.

Reflection should be avoided as much as possible.

Reply:
@romani, replaced reflection with additional audit call. Please note, that `XMLLogger` prints additional `<?xml version...`, `<checkstyle...` and `</checkstyle>` tags on the second call. The full output for this test is:

```
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="8.2-SNAPSHOT">
  <file name="Test.java">
    <error line="1" column="1" severity="error" message="key" source="com.puppycrawl.tools.checkstyle.XMLLoggerTest"/>
  </file>
</checkstyle>
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="8.2-SNAPSHOT">
</checkstyle>
```

I am not sure if this is an expected behavior. There are no tests for re-using the same `XMLLogger` for another audit report. Probably, we could just remove the `fileMessages.clear();` line from the `XMLLogger`

Reply:
@soon ,

if you use the same outputstream - it is expected that wrapper content is duplicated.
Only user know how to manage outputstreams. in UT, it is better to not reuse streams and recheck that results of 1st run is identical to output from 2nd run.

Reply:
@romani I've added additional output stream, please review